repositories/src/main/scala/cromwell/pipeline/datastorage/formatters/package.scala

```diff
@@ -0,0 +1,37 @@
+package cromwell.pipeline.datastorage
+
+import play.api.libs.json.{ Format, JsResult, JsValue, Json }
+import cats.Show
+import cromwell.pipeline.model.validator.Wrapped
+import cromwell.pipeline.model.wrapper.{ Name, Password, UserEmail, UserId, VersionValue }
+
+import scala.language.implicitConversions
+import cats.data.Validated.{ Invalid, Valid }
+import cats.implicits._
+import cromwell.pipeline.model.validator.MicrotypeException.MicrotypeValidationException
+import cromwell.pipeline.model.validator.WrappedFactoryPattern
+
+package object formatters {
+  private def wrapperFormat[T, E, W <: Wrapped[T]](wrapperFactory: WrappedFactoryPattern[T, E, W])(
+    implicit show: Show[E],
+    format: Format[T]
+  ): Format[W] =
+    new Format[W] {
+      override def reads(json: JsValue): JsResult[W] = json
+        .validate[T]
+        .map[W](
+          value =>
+            wrapperFactory.from(value) match {
+              case Valid(wrapped) => wrapped
+              case Invalid(e)     => throw MicrotypeValidationException("Microtype is not valid: " + e)
```

Comment:
Use `JsError` instead.
You should avoid throwing exceptions wherever possible if the return type allows you to do it.

Reply:
You advised me to use from(). from() has two projections: valid and invalid, what can I do with it? (As I understand, I have two validations here: one is json.validate, another is WrappedFactory.validate(), first returns JsSuccess/JSError, second returns ValidationResult. Should I override json.validate to inject WrappedFactory.validate() there? How can I do this, if from() contains both validate() and create() making it incapsulated/protcted?)

Reply:
No need to override anything. Take a look at a suggestion below.

Reply:
done